chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -1140,6 +1155,58 @@ impl Ledger {
         Ok(self)
     }
 
+    pub fn apply_encrypted_vote_tally<'a>(
+        self,
+        tally: &certificate::EncryptedVoteTally,
+        bad: &TransactionBindingAuthData<'a>,
+        sig: certificate::TallyProof,
+    ) -> Result<Self, Error> {
+        if sig.verify(tally.tally_type(), bad) == Verification::Failed {
+            return Err(Error::VoteTallyProofFailed);
+        }
+
+        // TODO: Anything to do here if not just verifying?
+        //
+        // let stake = StakeControl::new_with(&self.accounts, &self.utxos);
+        //
+        // let mut actions = Vec::new();
+        //
+        // let mut f = |action: &VoteAction| actions.push(action.clone());
+
+        // self.votes = self.votes.apply_committee_result(
+        //     self.date(),
+        //     &stake,
+        //     &self.governance,
+        //     tally,
+        //     sig,
+        //     &mut f,
+        // )?;
+
+        // for action in actions {
+        //     match action {
+        //         VoteAction::OffChain => {}
+        //         VoteAction::Treasury {
+        //             action: TreasuryGovernanceAction::NoOp,
+        //         } => {}
+        //         VoteAction::Treasury {
+        //             action: TreasuryGovernanceAction::TransferToRewards { value },
+        //         } => {
+        //             let value = self.pots.draw_treasury(value);
+        //             self.pots.rewards_add(value)?;
+        //         }
+        //         VoteAction::Parameters { action } => {
+        //             if self.governance.parameters.logs_register(action).is_err() {
+        //                 unimplemented!("the action was already recorded for this epoch")
+        //             } else {
+        //                 // nothing
+        //             }
+        //         }
+        //     }
+        // }
+
+        Ok(self)
```

Comment:
I don't really know what to do here if not just verifying it. @eugene-babichenko , @rinor

Reply:
You need to build the encrypted tally state and store it (I guess somewhere in the proposal manager).

Weights should be retrieved just like we do that for public tally.

https://github.com/input-output-hk/chain-libs/blob/master/chain-vote/src/lib.rs#L97

Reply:
@eugene-babichenko , where are we retrieving the weights for the public tally?

Reply:
How about the actions? Just to confirm but we really do not need them for the private tally right? @rinor

Reply:
Yeah, actions should be triggered on VoteTally certificate. Probably we should call this one VoteEncryptedTally to avoid confusion.